drivers/net/ethernet/intel/idpf/idpf_eth_idc.c

```diff
@@ -0,0 +1,431 @@
+ * @adapter: Idpf private structure
+ * @v_id: vport id to translate
+ *
+ * Returns dev_info matching v_id, NULL if not found.
+ */
+static struct idpf_eth_idc_dev_info *
+idpf_eth_idc_vid_to_dev_info(struct idpf_adapter *adapter,
+			     u32 v_id)
+{
+	struct idpf_eth_idc_dev_info *eth_info;
+	int i;
+
+	for (i = 0; i < adapter->default_vports; ++i) {
+		if (!adapter->adevs[i])
+			continue;
+		eth_info = &adapter->adevs[i]->eth_info;
+		if (v_id == eth_info->vport_id)
+			return eth_info;
+	}
+
+	return NULL;
+}
+
+/**
+ * idpf_eth_idc_dispatch_event - Called by Main Driver to send event
+ * @adapter: Idpf adapter private structure
+ * @event_type: Event type for all or single ports
+ * @event_code: Event code
+ * @event_data: Event data related to event code
+ */
+void idpf_eth_idc_dispatch_event(struct idpf_adapter *adapter,
+				 enum idpf_eth_idc_event_type event_type,
+				 enum idpf_eth_idc_event_code event_code,
+				 void *event_data)
+{
+	struct idpf_eth_idc_auxiliary_driver *eth_idc_drv;
+	struct idpf_eth_idc_dev_info *eth_info;
+	struct idpf_eth_idc_event event;
+	struct virtchnl2_event *v2e;
+	u32 vport_id;
+	int i;
+
+	if (!adapter->adevs)
+		return;
+
+	event.event_code = event_code;
+	event.event_data = event_data;
+	eth_idc_drv = idpf_eth_idc_get_driver();
+	switch (event_type) {
+	case IDPF_ETH_IDC_EVENT_ALL_VPORTS:
+		for (i = 0; i < adapter->default_vports; ++i) {
+			if (!adapter->adevs[i])
+				continue;
+
+			eth_info = &adapter->adevs[i]->eth_info;
+			eth_idc_drv->event_handler(eth_info, &event);
+		}
+		break;
+
+	case IDPF_ETH_IDC_EVENT_SINGLE_VPORT:
+		switch (event.event_code) {
+		case IDPF_ETH_IDC_EVENT_LINK_CHANGE:
+			v2e = (struct virtchnl2_event *)event.event_data;
+			/* vport_id indexes the dev_info instance */
+			vport_id = le32_to_cpu(v2e->vport_id);
+			eth_info =  idpf_eth_idc_vid_to_dev_info(adapter,
+								 vport_id);
+			if (eth_info)
+				eth_idc_drv->event_handler(eth_info, &event);
+			break;
+
+		default:
+			break;
+		}
+		break;
+
+	default:
+		break;
+	}
+}
+
+/**
+ * idpf_eth_idc_init_shared - Initialize ethernet shared struct
+ * @eth_shared: Ethernet and main apdater shared struct
+ *
+ * Returns 0 on success, negative on failure
+ */
+int idpf_eth_idc_init_shared(struct idpf_eth_shared *eth_shared)
+{
+	/* IDC ops */
+	eth_shared->eth_idc_ops.event_send = idpf_eth_idc_event_send;
+	eth_shared->eth_idc_ops.virtchnl_send = idpf_eth_idc_virtchnl_send;
+	eth_shared->eth_idc_ops.intr_reg_init = idpf_eth_idc_intr_reg_init;
+	eth_shared->eth_idc_ops.intr_init_vec_idx =
+		idpf_eth_idc_intr_init_vec_idx;
+	eth_shared->eth_idc_ops.req_rel_vec_idx =
+		idpf_eth_idc_req_rel_vector_indexes;
+
+	return 0;
+}
+
+/**
+ * idpf_eth_idc_deinit_shared - Initialize ethernet shared struct
+ * @eth_shared: Ethernet and main apdater shared struct
+ */
+void idpf_eth_idc_deinit_shared(struct idpf_eth_shared *eth_shared)
+{
+	eth_shared->hw_addr = NULL;
```

Comment:
This single line function should be avoided. Please make it NULL in calling function itself.

Reply:
Resolved.